corehq/apps/smsbillables/models.py

```diff
@@ -328,33 +350,43 @@ def create(cls, message_log, multipart_count=1):
         return billable
 
     @classmethod
-    def _get_gateway_fee(cls, backend_api_id, backend_instance, phone_number, direction, couch_id):
+    def _get_gateway_fee(cls, backend_api_id, backend_instance,
+                         phone_number, direction, couch_id, backend_message_id):
         country_code, national_number = get_country_code_and_national_number(phone_number)
         is_gateway_billable = backend_instance is None or _sms_backend_is_global(backend_instance)
 
         if is_gateway_billable:
-            gateway_fee = SmsGatewayFee.get_by_criteria(
-                backend_api_id,
-                direction,
-                backend_instance=backend_instance,
-                country_code=country_code,
-                national_number=national_number,
-            )
+            is_twilio_message = backend_api_id == SQLTwilioBackend.get_api_id()
+            if is_twilio_message:
+                twilio_charges = cls._get_twilio_charges(
+                    backend_message_id, direction, couch_id
+                )
+                gateway_fee = twilio_charges.gateway_fee
+                direct_gateway_fee = twilio_charges.twilio_gateway_fee
+            else:
+                gateway_fee = SmsGatewayFee.get_by_criteria(
+                    backend_api_id,
+                    direction,
+                    backend_instance=backend_instance,
+                    country_code=country_code,
+                    national_number=national_number,
+                )
+                direct_gateway_fee = None
             if gateway_fee:
```

Comment:
I'd like to continue to store a record with each billable of how the charge was decided on, which the gateway fee clarifies.  For example, if another gateway was added that needed to use ```direct_gateway_fee```, I'd like to be able to determine via the smsbillables models how the charge was determined (separately from the sms models).